runtime/src/iree-amd-aie/driver/xrt-lite/driver.cc

```diff
@@ -0,0 +1,133 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/driver/xrt-lite/api.h"
+#include "iree-amd-aie/driver/xrt-lite/util.h"
+
+#define IREE_HAL_XRT_LITE_DEVICE_ID_DEFAULT 0
```

Comment:
I don't know; this is used to populate 

```c++
  static const iree_hal_device_info_t device_infos[1] = {
      {
          .device_id = IREE_HAL_XRT_LITE_DEVICE_ID_DEFAULT,
          .name = iree_string_view_literal("default"),
      },
  };
```

and I'm not sure what/where `device_id` is used - I mean I can grep and it's used for retrieval/association of various things. 

What I don't have clear in my head right this moment is how this will look when we/if we try to partition - my guess is these IDs capture/correspond to something like multiple GPUs, which would be one way to implement/model multiple partitions running concurrently on one AIE device. But then again maybe not :shrug:.

Reply:
yea https://github.com/nod-ai/iree-amd-aie/blob/977edb6f9dbbb2ca8d97d7aaeb3ce96c2288d254/runtime/src/iree-amd-aie/driver/xrt-lite/shim/linux/kmq/amdxdna_accel.h#L413 - that's the "query" struct and I can't find right now where/how the setting of the `start_col` works but it's in there somewhere (or possibly it's one of their hidden features - i.e., only visible in the TurinAI/amd-aie repo).

Reply:
a little experiment: I added another entry to `device_infos` and I got this

```
$ iree-run-module --xrt_lite_n_core_rows=4 --xrt_lite_n_core_cols=4 --dump_devices

# ============================================================================
# Enumerated devices for driver 'xrt-lite'
# ============================================================================

# ===----------------------------------------------------------------------===
# --device=xrt-lite://
#   default
# ===----------------------------------------------------------------------===
# ===----------------------------------------------------------------------===
# --device=xrt-lite://
#   default-2
# ===----------------------------------------------------------------------===
```

**useful note**: `--xrt_lite_n_core_rows --xrt_lite_n_core_cols` have have to be passed first because of how [flag parsing works](https://github.com/iree-org/iree/blob/dc6f0cd5f4a784531bde3dcac48f2effc28c4224/runtime/src/iree/base/internal/flags.c#L526)